api/src/main/java/org/pmiops/workbench/cohortbuilder/CohortBuilderServiceImpl.java

```diff
@@ -492,8 +492,8 @@ public List<Criteria> findStandardCriteriaByDomainAndConceptId(String domain, Lo
   }
 
   @Override
-  public synchronized MultiKeyMap findAllDemographicsMap() {
-    MultiKeyMap demoMap = MultiKeyMap.multiKeyMap(new LRUMap<>());
+  public synchronized MultiKeyMap<Object, String> findAllDemographicsMap() {
+    MultiKeyMap<Object, String> demoMap = MultiKeyMap.multiKeyMap(new LRUMap<>());
```

Comment:
unfortunate that the 2 keys can't be of different types.  Object seems to be the best we can do.

Reply:
we cannot use generics with MultiKeyMap, and the requirement is to have 2 Keys [ConceptId (long), CriteriaType (enum)]. We could convert them to a combined String, but that beats the purpose

Reply:
@freemabd suggested to use HashBasedTable as it will satisfy the requirements. I am looking into it.